### PR TITLE
Moving rsyslog help article

### PR DESCRIPTION
**Pull request by z0r1k**
# What changed, and why it matters

Closes https://github.com/aiven/devportal/issues/828

### CODE REVIEW

**Comment by lornajane** [comment]
I moved the DataDog stuff already as well and I would be so happy to get this content moved! Great to see this work

**Comment by lornajane** [comment]
I think it would be even better to have focused articles per integration, as I did for DataDog. I see a lot of questions about this where people haven't been able to find our resources, or think that we don't support the integration they are looking for. We can keep a generic rsyslog article as well, in case anyone needs something we haven't specifically documented.

Reviewers: as with all the docs, we'll need to test all the steps before we merge ...

**Comment by runwuf** [comment]
Got it @lornajane
For the integrations in this article, I can confirm datadog and coralogix integration works, I have validated these two integration in the past.

**Comment by TibsAtWork** [comment]
Our fault for not being specific on this, but in general we prefer to rebase a branch on main rather than merge main into a branch.
This needs a little bit more collaboration between people, but gives a cleaner history.
We're very happy to do it for you if you don't want to fight git.

**Comment by runwuf** [comment]
I will work with @dragonlancewu to rebase this on main.

**Comment by runwuf** [comment]
@TibsAtWork - we rebased from the main and it looks good now.
@z0r1k - I've overwritten your merged main commit so I can do a clean rebase, you'll have to reset your local branch with remote to keep them in-sync.

File: docs/integrations/rsyslog/index.rst
Comment (laysauchoa) [inline docs/integrations/rsyslog/index.rst:158]: I believe this should be part of the datadog section instead of embedded here and here maybe a link to there, what do you think?

File: docs/integrations/rsyslog/index.rst
Comment (laysauchoa) [inline docs/integrations/rsyslog/index.rst:81]: I'd suggest to remove this image and just add leave it with instructions. You can see step 5 here: https://developer.aiven.io/docs/integrations/datadog/datadog-metrics.html

File: docs/integrations/rsyslog/index.rst
Comment (runwuf) [reply docs/integrations/rsyslog/index.rst:157]: I kept this section as it is using `avn` cli to add the service integration but added a link to the datadog integration
```:doc:`Further details on send metrics and logs to Datadog </docs/integrations/datadog/index>```
Let me know if you have better suggestions.

File: docs/integrations/rsyslog/index.rst
Comment (runwuf) [reply docs/integrations/rsyslog/index.rst]: Removed this one as per suggestion.  I have also removed the other old screenshots and replaced with new screenshots from new Aiven console UI.

File: docs/integrations/rsyslog/index.rst
Comment (TibsAtWork) [inline docs/integrations/rsyslog/index.rst:213]: Is `programname` meant to be replaced by specific text?

File: docs/integrations/rsyslog/index.rst
Comment (TibsAtWork) [inline docs/integrations/rsyslog/index.rst:230]: Would the user expect backslashes around `key`?

**Review by TibsAtWork — COMMENTED** [review]
I've done some reformatting of the text in the "Some formatting changes" commit.

I still need to work out what to do about the various product names (to stop the linting from complaining that they are unrecognised words). These are warnings (so not stopping the build), but I'd like to get them fixed. I'll *probably* add them to the dictionary - that's definitely my job, not yours 😄

I suspect it will also be a good idea to link to product pages in the text, but let's address that separately.

Let me know if you want help with the file reorganisation.

File: docs/integrations/rsyslog/index.rst
Comment (TibsAtWork) [reply docs/integrations/rsyslog/index.rst]: ..I fixed it

File: docs/integrations/rsyslog/index.rst
Comment (runwuf) [reply docs/integrations/rsyslog/index.rst]: Removed this to avoid confushion.

File: docs/integrations/rsyslog/index.rst
Comment (runwuf) [reply docs/integrations/rsyslog/index.rst]: Nice catch, it should have been a variable `% programname%`

**Comment by runwuf** [comment]
@TibsAtWork thanks for your review, please let me know what else needs to be fixed.

**Comment by TibsAtWork** [comment]
I'll sort out the product names and then look over the article again.

File: docs/integrations/rsyslog.rst
Comment (runwuf) [reply docs/integrations/rsyslog.rst]: Thanks for pointing this out @TibsAtWork , I have updated to use license key instead.

**Comment by runwuf** [comment]
Thanks for all your work @TibsAtWork !
I've added a small ToC section on top of the page as I think it's useful to see what integrations are in this article without having to scroll through the page (I attempted to create links to anchor on the same file from _toc.yml but it looks like not supported).  I've also rearranged integrations in alphabetical orders.  What do you think?

**Comment by TibsAtWork** [comment]
We shouldn't need the extra table of contents, because there should already be one at the right hand side:

Screenshot 2022-07-13 at 10 47 30

That goes away if the browser window is narrow enough (as does the main sidebar on the left, eventually).
Otherwise, if you're not seeing the "normal" right hand side ToC, let us know, as we probably need to fix that.

**Comment by TibsAtWork** [comment]
I've made the following changes:

* Correct the link in the rsyslog overview page
* Remove the explicit ToC again, since there should automatically be one at the right of the page
* Add `alt` text to the images to make the vale linter happy
* Make the intro sentence for each integration more consistent, which also removes the "name/link standing by itself" I'd introduced
* Indent all the code samples the same - I went with 4 spaces rather than none, as that seemed to be more common
* Change the link to the `datadog.html` page to make vale happy

@runwuf I think this may now be good to go, for editorial purposes.

Have all of the examples been checked to make sure they're accurate / "work on this machine as well"?

File: docs/integrations/rsyslog.rst
Comment (lornajane) [inline docs/integrations/rsyslog.rst:163]: I'm not sure what this is doing here but I think we're removing the section anyway so perhaps it doesn't matter

**Review by lornajane — DISMISSED** [review]

**Comment by lornajane** [comment]
I'd still really like these as separate articles, on the basis that most users will be implementing only one at any one time, even if we think of them from the back end as a single feature. However I don't want to block progress, so if we merge as-is, please open a followup issue to create dedicated guides for each platform, as we did for Datadog.

File: docs/integrations/rsyslog.rst
Comment (TibsAtWork) [reply docs/integrations/rsyslog.rst:144]: @runwuf can I leave that to you to sort out?

**Comment by TibsAtWork** [comment]
I agree with @lornajane about treating this as a directory of specific files in the future, which will actually be easier with the new ways we organise things. However I also think it's worth doing this "quick fix" now, and also it will actually make the redirection from the original help easier to use.

I can raise an issue for more work once we've merged.

**Comment by runwuf** [comment]
@TibsAtWork  I had no idea of the existence of the toc on the right hand side until today!!! because I have always been using a narrow browser window... 🤦‍♂️
btw, I didn't invent `:class: this-will-duplicate-information-and-it-is-still-useful-here ` when I didn't define this class, it shows up as an alert on the page.

I can confirm Coralogix (I wrote this config) and Datadog works.

The following I have not tested personally, I'll drop a note to our SA team see if others have tested any of them:
- Loggly
- Mezmo (LogDNA)
- New Relic
- Papertrail
- Sumo Logic

**Comment by runwuf** [comment]
@TibsAtWork I validated and updated ALL of these rsyslog integration configs.

**Comment by TibsAtWork** [comment]
@runwuf Thanks very much for all the work on this!

I've moved the "for this remote syslog service you need *these* special considerations" text around a bit so all of the entries have a similar layout, with the setup information before the example (a colleague explained to me the other day that this makes it much more likely people will notice them). I've also done some more reStructuredText tidying, and made the vale linter happy by changing how a link is done.

Basically I think this is now all done, and shall approve and merge it.

I've raised issue https://github.com/aiven/devportal/issues/1148 to look at splitting it into sub-pages later on.

**Review by TibsAtWork — APPROVED** [review]
Approved

**Pull request merged**
